log/testutil.go

```diff
@@ -0,0 +1,122 @@
+package log
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+)
+
+// FieldValidator facilitates testing of Span.Log*() implementations.
+//
+// FieldValidator SHOULD ONLY BE USED FOR TESTS. It is not intended for
+// production code.
+//
+// Usage:
+//
+//     fv := log.NewFieldValidator(t, someLogStructure.Fields)
+//     fv.
+//         ExpectNextFieldEquals("key1", reflect.String, "some string value").
+//         ExpectNextFieldEquals("key2", reflect.Uint8, "255")
+//
+// FieldValidator satisfies the log.Encoder interface and thus is able to
+// marshal log.Field instances (which it takes advantage of internally).
+type FieldValidator struct {
```

Comment:
See https://github.com/opentracing/basictracer-go/pull/37/files#diff-94fbb48b23e68ffcc13ee7a6aa73773aR99 for an example of how this can be used.